var/spack/repos/builtin/packages/netcdf-cxx4/package.py

```diff
@@ -71,9 +91,21 @@ def configure_args(self):
         else:
             config_args.append('--without-pic')
 
-        if '+doxygen' in self.spec:
+        if '+doc' in self.spec:
             config_args.append('--enable-doxygen')
         else:
             config_args.append('--disable-doxygen')
 
         return config_args
+
+    @when('@:4.3.1')
+    def build(self, spec, prefix):
+        if spec.satisfies("platform=darwin"):
+            # If on macos, rename the file "VERSION" so it doesn't collide with the
+            # c++ include file (named "version"). Note this collision occurs since macos
+            # uses a case-insensitive file system.
+            #
+            # Unidata has fixed this in their development track:
+            #   https://github.com/Unidata/netcdf-cxx4/commit/41c0233cb964a3ee1d4e5db5448cd28d617925fb
+            os.rename('VERSION', 'config.VERSION')
+        super(NetcdfCxx4, self).build(spec, prefix)
```

Comment:
could you, please, check whether this can be replaced with (also for version `4.3.0`):
```python
patch('https://github.com/Unidata/netcdf-cxx4/commit/e7cc5bab02cf089dc79616456a0a951fee979fe9.patch',
      sha256='ba466b887c7ad2710e6d394046bc0742a2f4043d98bd21d5b08f33dc9d7f392d',
      when='@:4.3.1 platform=darwin')
```
I have used https://github.com/Unidata/netcdf-cxx4/commit/e7cc5bab02cf089dc79616456a0a951fee979fe9 because that's the commit in the `master` upstream.

Reply:
Ok, I got this - will update the sha hash in my next test:
```
==> Error: ChecksumError: sha256 checksum failed for /Users/heinzell/work/jedi-stack/spack-stack-new-joint/spack-stack-develop-20220215/cache/build_stage/spack-stage-u032oec1/e7cc5bab02cf089dc79616456a0a951fee979fe9.patch
    Expected ba466b887c7ad2710e6d394046bc0742a2f4043d98bd21d5b08f33dc9d7f392d but got 4ddf6db9dc0c5f754cb3d68b1dbef8c385cf499f6e5df8fbccae3749336ba84a
```

Reply:
My bad, should be `4ddf6db9dc0c5f754cb3d68b1dbef8c385cf499f6e5df8fbccae3749336ba84a` indeed.

Reply:
You might need to run `spack clean -m` after the update and before the attempt to build.

Reply:
Sure, I have learned (the hard way) that making spack forget isn't that easy ... The patch works just fine.